modules/ROOT/pages/queries/expressions.adoc

```diff
@@ -0,0 +1,79 @@
+= Cypher expressions
+:description: This page explains which expressions are allowed in Cypher.
+
+This page contains examples of allowed expressions in Cypher.
+
+[NOTE]
+====
+Most expressions in Cypher evaluate to `null` if any of their inner expressions are `null`.
+Notable exceptions are the operators `IS NULL`, `IS NOT NULL`, and the xref:values-and-types/type-predicate.adoc[type predicate expressions].
+====
+
+[[general]]
+== General
+
+* A variable: `n`, `x`, `rel`, `myFancyVariable`, `++`A name with special characters in it[]!`++`.
+* A property: `n.prop`, `x.prop`, `rel.thisProperty`, `++myFancyVariable.`(special property name)`++`.
+* A dynamic property: `n["prop"]`, `rel[n.city + n.zip]`, `map[coll[0]]`.
+* A parameter: `$param`, `$0`.
+* A list of expressions: `['a', 'b']`, `[1, 2, 3]`, `['a', 2, n.property, $param]`, `[]`.
+* A function call: `length(p)`, `nodes(p)`.
+* An aggregate function call: `avg(x.prop)`, `+count(*)+`.
+* A path-pattern: `+(a)-[r]->(b)+`, `+(a)-[r]-(b)+`, `+(a)--(b)+`, `+(a)-->()<--(b)+`.
+* An operator application: `1 + 2`, `3 < 4`.
+* A subquery expression: `COUNT {}`, `COLLECT {}`, `EXISTS {}`, `CALL {}`.
+* A regular expression: `a.name =~ 'Tim.*'`.
+* A `CASE` expression.
+
+[NOTE]
+====
+Using regular expressions with unsanitized user input may make your application vulnerable to Cypher injection.
+Consider using xref:syntax/parameters.adoc[parameters] instead.
+====
+
+[[numerical]]
+== Numerical
+
+* A numeric (`INTEGER` or `FLOAT`) literal: `13`, `-40000`, `3.14`.
+* A numeric (`INTEGER` or `FLOAT`) literal in scientific notation: `6.022E23`.
+* A hexadecimal `INTEGER` literal (starting with `0x`): `0x13af`, `0xFC3A9`, `-0x66eff`.
+* An octal `INTEGER` literal (starting with `0o`): `0o1372`, `-0o5671`.
+* A `FLOAT` literal: `Inf`, `Infinity`, `NaN`.
+
+[NOTE]
+====
+Any numeric literal may contain an underscore `_` between digits.
+There may be an underscore between the `0x` or `0o` and the digits for hexadecimal and octal literals.
+====
+
+[[string]]
+== String
+
+* A `STRING` literal: `'Hello'`, `"World"`.
+* A case-sensitive `STRING` matching expression: `a.surname STARTS WITH 'Sven'`, `a.surname ENDS WITH 'son'` or `a.surname CONTAINS 'son'`.
+
+[[expressions-string-literals]]
+=== String literal escape sequences
+
+String literals can contain the following escape sequences:
+
+[options="header", cols=">1,<2"]
+|===================
+|Escape sequence|Character
+|`\t`|Tab
+|`\b`|Backspace
+|`\n`|Newline
+|`\r`|Carriage return
+|`\f`|Form feed
+|`\'`|Single quote
+|`\"`|Double quote
+|`\\`|Backslash
+|`\uxxxx`|Unicode UTF-16 code point (4 hex digits must follow the `\u`)
+|===================
+
+[[boolean]]
+== Boolean
+
+* A `BOOLEAN` literal: `true`, `false`.
+* A predicate expression is an expression that returns a `BOOLEAN` value, `true`, `false` or `null`: `a.prop = 'Hello'`, `length(p) > 10`, `a.name IS NOT NULL`.
```

Comment:
```suggestion
* A predicate expression (i.e. an expression that returns a `BOOLEAN` value): `a.prop = 'Hello'`, `length(p) > 10`, `a.name IS NOT NULL`.
```

Reply:
@gem-neo4j `null` belongs to all types, am I correct?

Reply:
Yep, it does :)

Reply:
Yeah that's a tricky one. (And we transition a bit from being pedagogical and developer friendly to reading into people's minds.) 
What I don't like of mentioning it in booleans only is that it leads to think that `null` _only_ belongs to booleans. For me it would definitely make sense to have `null` as a separate list item in all sections.

--

PS. What I didn't like about the wording of predicate expressions is that it felt out of place. In my mind, the section reads:
> A boolean is ...
> - ... a boolean literal
> - ... a predicate expression is an expression that returns a boolean value
> - ... label and rel type expressions

and I thought ... why is the second point there (mostly due to the double `is`)? It took me a second to realize that it was not describing something _else_, but that, while it was introducing something else, that object was also part of the list.